tools/tests-consolidated/security/input-validation.test.ts

```
@@ -0,0 +1,797 @@
+    if (maxLength !== undefined && input.length > maxLength) {
+      return false;
+    }
+    
+    // Check for allowed characters
+    if (allowedChars !== undefined && !allowedChars.test(input)) {
+      return false;
+    }
+    
+    // Check for dangerous characters
+    const dangerousPatterns = [
+      /\x00/, // Null byte
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)